packages/repository/src/__tests__/unit/repositories/has-many-through-repository-factory.unit.ts

```diff
@@ -0,0 +1,152 @@
+// Copyright IBM Corp. 2017,2018,2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Getter} from '@loopback/context';
+import {createStubInstance, expect} from '@loopback/testlab';
+import {
+  DefaultCrudRepository,
+  Entity,
+  HasManyThroughDefinition,
+  ModelDefinition,
+  RelationType,
+  createHasManyThroughRepositoryFactory,
+  juggler,
+} from '../../..';
+
+describe('createHasManyThroughRepositoryFactory', () => {
```

Comment:
I am proposing to add the following two test cases to this test suite:

- `it('rejects relations with missing "through"')`
- `it('rejects relations with "through" that is not a type resolver')`

Reply:
Good idea

Reply:
I have added those tests.